tutorials/preprocessing/plot_35_artifact_correction_regression.py

```diff
@@ -0,0 +1,142 @@
+not apply here -- see :ref:`tut-brainstorm-auditory` for more information.
+"""
+
+import os.path as op
+import numpy as np
+import mne
+
+data_path = mne.datasets.brainstorm.bst_auditory.data_path()
+raw_fname = op.join(data_path, 'MEG', 'bst_auditory', 'S01_AEF_20131218_01.ds')
+raw = mne.io.read_raw_ctf(raw_fname).crop(0, 60)
+raw.set_channel_types({'HEOG': 'eog', 'VEOG': 'eog', 'ECG': 'ecg'})
+raw.pick_types(meg=True, stim=True, misc=True,
+               eog=True, ecg=True, ref_meg=False).load_data()
+raw.info['bads'] = ['MLO52-4408', 'MRT51-4408', 'MLO42-4408', 'MLO43-4408']
+raw.filter(None, 40)
+decim = 12  # 2400 -> 200 Hz sample rate for epochs
+
+###############################################################################
+# Removing artifacts by regression
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# Artifacts whose time waveforms are accurately reflected by some reference
+# signal can be removed by regression. Two examples of this are EOG artifacts
+# captured by EOG channels, and ECG signals captured by ECG electrodes.
+#
+#
+# EOG
+# ~~~
+#
+# First let's epoch our data the normal way, using just event type 1 (standard)
+# for simplicity. Event timing is adjusted by comparing the trigger times on
+# detected sound onsets on channel UADC001-4408.
+
+events = mne.find_events(raw, stim_channel='UPPT001')
+sound_data = raw[raw.ch_names.index('UADC001-4408')][0][0]
+onsets = np.where(np.abs(sound_data) > 2. * np.std(sound_data))[0]
+min_diff = int(0.5 * raw.info['sfreq'])
+diffs = np.concatenate([[min_diff + 1], np.diff(onsets)])
+onsets = onsets[diffs > min_diff]
+assert len(onsets) == len(events)
+events[:, 0] = onsets
+epochs = mne.Epochs(raw, events, event_id=1, decim=decim, preload=True)
+
+###############################################################################
+# Now let's regress our EOG signal and plot the original and processed data.
+# We do this by estimating the regression coefficients on data without the
+# evoked response, then use those coefficients to remove the EOG signal:
+
+plot_picks = ['meg', 'eog', 'ecg']
+evo_kwargs = dict(picks=plot_picks, spatial_colors=True,
+                  verbose='error')  # ignore warnings about spatial colors
+fig = epochs.average(picks=plot_picks).plot(**evo_kwargs)
+fig.suptitle('Auditory epochs')
+fig.tight_layout()
+
+epochs_no_ave = epochs.copy().subtract_evoked()
+_, betas = mne.preprocessing.regress_artifact(epochs_no_ave)
+epochs_clean, _ = mne.preprocessing.regress_artifact(epochs, betas=betas)
+fig = epochs_clean.average(picks=plot_picks).plot(**evo_kwargs)
+fig.suptitle('Auditory epochs, EOG regressed')
+fig.tight_layout()
+del epochs, epochs_clean
+
+###############################################################################
+# The effect is subtle in these evoked data. You can see that a bump toward
+# the end of the window has had its amplitude decreased.
+#
+# The effect is even clearer if we create epochs around our (autodetected) EOG
```

Comment:
```suggestion
# The effect is clearer if we create epochs around our (autodetected) EOG
```

Reply:
I have a hard time agreeing that the effect is clear in the current plots. Can we zoom in on the second, larger blink (around 21 seconds) and show fewer MEG channels (so that the blink perturbation looks bigger)?  I'd also consider showing the 2 raw plots side-by-side.  I'm neutral on whether showing the two EOG epochs plots adds enough to be worth it; understanding the improvement relies on noticing the different y-axis limits, so I would either fix the limits across the two `mag` panels, or call attention to the changed y-axis limits in the text.